Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM eclipse-temurin:11-alpine AS base
+FROM eclipse-temurin:17 AS base
```

Comment:
Can we use the [`eclipse-temurin:17-alpine`](https://hub.docker.com/layers/eclipse-temurin/library/eclipse-temurin/17-alpine/images/sha256-b30fa3ce4323ce037cb95fd2729dd4662d86f0ee2986452527cc645eaf258a1d?context=explore) tag to reduce the overall size of the image?

Reply:
Are there any tests or test procedures to validate that Marquez works for a given image/JDK? It is probably worth having at least a manual testing strategy to validate a change this large.

Reply:
Yeah, it's a big jump for sure. I would use:

```
$ ./docker/up.sh --build --seed
```

This will ensure the images are built from source and seeding the backend will still allow for displaying metadata via the UI.

Reply:
@wslulciuc the problem with the alpine images is that they don't seem to be publishing for multiple CPU architectures. I.e. if I use the alpine image, it won't work on the M1/ARM chips. You can see [for `17-alpine` the only supported CPU is `amd64`.](https://github.com/docker-library/official-images/blob/master/library/eclipse-temurin#L280-L282)
Whereas, [`17` has a full list of supported CPU architectures, inclusive of `arm` and `amd`.](https://github.com/docker-library/official-images/blob/master/library/eclipse-temurin#L287-L292).

Reply:
I've run `./docker/up.sh --build --seed` and recorded a gif, of the build process and the subsequet successful use of Marquez. The gif is 62mbs though, so I can't upload it here. Sharing a screenshot instead.

<img width="1437" alt="image" src="https://user-images.githubusercontent.com/6220504/147984801-500862ba-3769-4e1e-9572-361947572d28.png">

Let me know if as reviewers the gif recording of the build + running would be useful and I'll upload it somewhere and link to it.

Reply:
> the problem with the alpine images is that they don't seem to be publishing for multiple CPU architectures

Ah makes sense, thanks for providing the links, @ucg8j.

Reply:
> Are there any tests or test procedures to validate that Marquez works for a given image/JDK? It is probably worth having at least a manual testing strategy to validate a change this large.

@collado-mike I think if:

1. The CI build pass
2. The CI tests pass
3. Locally, the API and UI run with no issues

Then, I think it's "safe" to merge. Of course, all Gradle plugins will be validated. There's also [this](https://blogs.oracle.com/javamagazine/post/its-time-to-move-your-applications-to-java-17-heres-why-and-heres-how) that outlines the migration to Java 17. Thoughts?

Reply:
As is, the tests are running in CI with JDK 11. Only the Docker build here is using JDK 17 and it calls `shadowJar`, which doesn't exercise the tests. I think we need to change the CircleCI config to also use JDK 17 to validate the tests pass.

Reply:
I've had a go at including this in the PR here https://github.com/MarquezProject/marquez/pull/1819/commits/63685ad476118ad052ea8254ff5737a4f287430f

Reply:
So the [CI check fails on api:spotlessJava step 'google-java-format'.](https://app.circleci.com/pipelines/github/MarquezProject/marquez/5471/workflows/d81041d8-39c2-4766-b9c5-59b5b827541a/jobs/41284/parallel-runs/0/steps/0-105)

```
Caused by: java.lang.IllegalAccessError: class com.google.googlejavaformat.java.JavaInput (in unnamed module @0x7711251f) cannot access class com.sun.tools.javac.parser.Tokens$TokenKind (in module jdk.compiler) because module jdk.compiler does not export com.sun.tools.javac.parser to unnamed module @0x7711251f
```

Solution seems to be:
```
java \
  --add-exports jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED \
  --add-exports jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED \
  --add-exports jdk.compiler/com.sun.tools.javac.parser=ALL-UNNAMED \
  --add-exports jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED \
  --add-exports jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED \
  -jar google-java-format-1.13.0-all-deps.jar <options> [files...]
```

[Source 1](https://github.com/google/google-java-format#jdk-16)
[Source 2](https://github.com/google/google-java-format/issues/538#issuecomment-780290917)

Which I think [would go here in the Marquez codebase](https://github.com/ucg8j/marquez/blob/fix-for-m1s/gradlew#L205) - I'm not super familiar with Java, so let me know if that seems right to you @collado-mike @wslulciuc

Reply:
Ah, you're one step ahead of me! Here are my thoughts: https://github.com/MarquezProject/marquez/pull/1819#pullrequestreview-843934829

Reply:
@wslulciuc  Snap! I *think* I've applied the appropriate gradle properties https://github.com/MarquezProject/marquez/pull/1819/commits/49182712119756c2935c8701b95bcfa212a6ba56 . But now getting this error

```
Unrecognized option: --add-exports
```

Let me know if you have ideas to get around that.

Reply:
https://github.com/MarquezProject/marquez/pull/1819#issuecomment-1005178983
Ah you're ahead of me! Will try it out now